src/components/rocm_smi/rocs.c

```diff
@@ -3975,3 +3995,74 @@ access_rsmi_dev_vendor_name(rocs_access_mode_e mode, void *arg)
     event->value = (int64_t) event->scratch;
     return PAPI_OK;
 }
+
+static int
+access_rsmi_dev_energy_count(rocs_access_mode_e mode, void *arg)
+{
+    ntv_event_t *event = (ntv_event_t *)arg;
+
+    /* This event is read-only. If a caller tries to WRITE, deny it. */
+    if (mode != ROCS_ACCESS_MODE__READ || mode != event->mode) {
+        return PAPI_ENOSUPP; /* not supported */
+    }
+
+    rsmi_status_t status;
+    uint64_t raw_count = 0ULL;
+    float    resolution = 0.0f;   /* in microjoules */
+    uint64_t timestamp = 0ULL;    /* in ns, if you care */
+
+    status = rsmi_dev_energy_count_get_p((uint32_t)event->device,
+                                         &raw_count,
+                                         &resolution,
+                                         &timestamp);
+    if (status != RSMI_STATUS_SUCCESS &&  status != RSMI_STATUS_INVALID_ARGS) {
+        /*
+         * Note: RSMI_STATUS_INVALID_ARGS can mean "power was null" if the device
+         * actually supports the call. If it’s truly unsupported, typically
+         * RSMI_STATUS_NOT_SUPPORTED is returned. Adjust logic if needed.
+         */
+        return PAPI_EMISC;
+    }
+
+    /*
+     * According to the doc, 'raw_count' * 'resolution' (in µJ) is the total
+     * energy consumed since the counter began accumulating. We store as
+     * an int64_t in microjoules. Watch for overflow if the accumulated
+     * energy is very large.
+     */
+    double total_uJ = (double)raw_count * (double)resolution;
+    if (total_uJ > INT64_MAX) {
+        /* clamp to avoid overflow */
+        total_uJ = INT64_MAX;
+    }
+
+    event->value = (int64_t)total_uJ;
+    return PAPI_OK;
+}
+
+static int
+access_rsmi_dev_current_socket_power(rocs_access_mode_e mode, void *arg)
+{
+    ntv_event_t *event = (ntv_event_t *)arg;
+
+    /* This event is read-only. If a caller tries to WRITE, deny it. */
+    if (mode != ROCS_ACCESS_MODE__READ || mode != event->mode) {
+        return PAPI_ENOSUPP; /* not supported */
+    }
+
+    rsmi_status_t status;
+    uint64_t power = 0ULL; /* in µW */
+
+    status = rsmi_dev_current_socket_power_get_p((uint32_t)event->device, &power);
```

Comment:
On a machine with 2 * AMD MI210s, I run into the error `RSMI_STATUS_NOT_SUPPORTED`, I saw that in documentation it is noted that this function is supported by newer devices. I was able to test on an MI300A and there I am able to monitor the `current_socket_power`. 

What devices have you tested this on? As I do not see a list from AMD that explicitly states which devices fall under this "newer" category.

Reply:
Got it! Thank you for the reply!